coreclr outerloop test build failure

https://dev.azure.com/dnceng/public/_build/results?buildId=576645&view=logs&j=8524357e-1450-5e10-a984-883a50af28d9&t=ede4d230-8340-53fe-3e6a-8c3ab5208a8a

```
test24701.cs(7,15): error CS0659: 'Data' overrides Object.Equals(object o) but does not override Object.GetHashCode() [F:\workspace\_work\1\s\src\coreclr\tests\src\Regressions\coreclr\GitHub_24701\test24701.csproj]
##[error]test24701.cs(7,15): error CS0659: 'Data' overrides Object.Equals(object o) but does not override Object.GetHashCode()
```

I believe due to https://github.com/dotnet/runtime/pull/34155 changing warnings to errors. Previously, this was a warning:

https://dev.azure.com/dnceng/public/_build/results?buildId=575509&view=logs&j=8524357e-1450-5e10-a984-883a50af28d9&t=ede4d230-8340-53fe-3e6a-8c3ab5208a8a
```
test24701.cs(7,15): warning CS0659: 'Data' overrides Object.Equals(object o) but does not override Object.GetHashCode() [F:\workspace\_work\1\s\src\coreclr\tests\src\Regressions\coreclr\GitHub_24701\test24701.csproj]
##[warning]test24701.cs(7,15): warning CS0659: (NETCORE_ENGINEERING_TELEMETRY=) 'Data' overrides Object.Equals(object o) but does not override Object.GetHashCode()
```

@jaredpar

--- a/src/coreclr/build-test.cmd
+++ b/src/coreclr/build-test.cmd
@@ -296,7 +296,7 @@ set __MsbuildErr=/flp2:ErrorsOnly;LogFile="%__BuildErr%"
 set __Logging='!__MsbuildLog!' '!__MsbuildWrn!' '!__MsbuildErr!'
 
 powershell -NoProfile -ExecutionPolicy ByPass -NoLogo -Command "%__RepoRootDir%\eng\common\msbuild.ps1" %__ArcadeScriptArgs%^
-  %__ProjectDir%\tests\build.proj -warnAsError:1 /t:BatchRestorePackages /nodeReuse:false^
+  %__ProjectDir%\tests\build.proj -warnAsError:0 /t:BatchRestorePackages /nodeReuse:false^
   /p:RestoreDefaultOptimizationDataPackage=false /p:PortableBuild=true^
   /p:UsePartialNGENOptimization=false /maxcpucount^
   %__SkipFXRestoreArg%^
@@ -355,7 +355,7 @@ for /l %%G in (1, 1, %__NumberOfTestGroups%) do (
 
     if not "%__CopyNativeTestBinaries%" == "1" (
         set __MSBuildBuildArgs=!__ProjectDir!\tests\build.proj
-        set __MSBuildBuildArgs=!__MSBuildBuildArgs! -warnAsError:1
+        set __MSBuildBuildArgs=!__MSBuildBuildArgs! -warnAsError:0
         set __MSBuildBuildArgs=!__MSBuildBuildArgs! /nodeReuse:false
         set __MSBuildBuildArgs=!__MSBuildBuildArgs! !__Logging!
         set __MSBuildBuildArgs=!__MSBuildBuildArgs! !TargetsWindowsMsbuildArg!
@@ -379,7 +379,7 @@ for /l %%G in (1, 1, %__NumberOfTestGroups%) do (
             goto     :Exit_Failure
         )
     ) else (
-        set __MSBuildBuildArgs=!__ProjectDir!\tests\build.proj -warnAsError:1 /nodeReuse:false !__Logging! !TargetsWindowsMsbuildArg! !__msbuildArgs!  !__PriorityArg! !__SkipFXRestoreArg! !__UnprocessedBuildArgs! "/t:CopyAllNativeProjectReferenceBinaries"
+        set __MSBuildBuildArgs=!__ProjectDir!\tests\build.proj -warnAsError:0 /nodeReuse:false !__Logging! !TargetsWindowsMsbuildArg! !__msbuildArgs!  !__PriorityArg! !__SkipFXRestoreArg! !__UnprocessedBuildArgs! "/t:CopyAllNativeProjectReferenceBinaries"
         echo Running: msbuild !__MSBuildBuildArgs!
         !__CommonMSBuildCmdPrefix! !__MSBuildBuildArgs!
 

